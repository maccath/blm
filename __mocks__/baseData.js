const baseData = {
  site: {
    siteMetadata: {
      author: "Site Author",
      description: "Site description",
      title: "Site Title",
    },
  },
  placeholderImage: {
    childImageSharp: {
      fluid: {
        src: "icon.png",
        srcSet: "icon.png 50w, icon.png 100w",
        sizes: "(max-width: 200px) 100vw, 200px",
        base64: "data:image/png;base64,iVBORw0KG",
        aspectRatio: 1,
      },
    },
  },
}

export default baseData
