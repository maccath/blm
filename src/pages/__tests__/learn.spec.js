import React from "react"
import renderer from "react-test-renderer"
import { useStaticQuery } from "gatsby"
import "jest-styled-components"
import baseData from "../../../__mocks__/baseData"

import Learn from "../learn"
import Layout from "../../components/layout"

beforeEach(() => {
  useStaticQuery.mockImplementation(() => baseData)
})

describe("Learn page", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Layout>
          <Learn />
        </Layout>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
