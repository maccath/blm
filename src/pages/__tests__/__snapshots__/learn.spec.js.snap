// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Learn page renders correctly 1`] = `
<div
  style={
    Object {
      "opacity": 1,
      "transition": "opacity 500ms ease-in-out",
    }
  }
>
  <div
    style={
      Object {
        "margin": "0 auto",
        "maxWidth": 960,
        "padding": "0 1.0875rem 1.45rem",
      }
    }
  >
    <main>
      <h1>
        Learn
      </h1>
      <p>
        There are a number of books and documentaries that explain white supremacy
and focus on anti-racist education.
      </p>
      <h2>
        Watch and listen
      </h2>
      <div>
        <iframe
          allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
          allowFullScreen=""
          frameBorder="0"
          height="315"
          src="https://www.youtube.com/embed/1SynR1NYcpo"
          title="What happens when I try to talk race with white people"
          width="560"
        />
      </div>
      <div>
        <iframe
          allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
          allowFullScreen=""
          frameBorder="0"
          height="315"
          src="https://www.youtube.com/embed/K6IXQbXPO3I"
          title="13TH | official trailer (2016) Netflix"
          width="560"
        />
      </div>
      <div>
        <iframe
          allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
          allowFullScreen=""
          frameBorder="0"
          height="315"
          src="https://www.youtube.com/embed/Mqrhn8khGLM"
          title="Explained | Racial Wealth Gap | FULL EPISODE"
          width="560"
        />
      </div>
      <ul>
        <li>
          <a
            href="https://www.netflix.com/gb/title/80091741"
          >
            Watch 13TH on Netflix
          </a>
        </li>
        <li>
          <a
            href="http://laylafsaad.com/good-ancestor-podcast"
          >
            The Good Ancestor Podcast
          </a>
        </li>
      </ul>
      <h2>
        Articles 
        &
         Opinions
      </h2>
      <ul>
        <li>
          <a
            href="https://www.theguardian.com/commentisfree/2020/jun/03/racism-george-floyd-britain-america-uk-black-people#maincontent"
          >
            The racism that killed George Floyd was built in Britain
          </a>
           - Afua Hirsch
        </li>
        <li>
          <a
            href="https://www.vogue.co.uk/arts-and-lifestyle/article/non-optical-ally-guide"
          >
            If You Are White And Anti-Racist, This Non-Optical Allyship Guide Is Required Reading
          </a>
           - Mireille Cassandra Harper
        </li>
      </ul>
      <h2>
        Anti-racist Book Club
      </h2>
      <ul>
        <li>
          <a
            href="https://www.hive.co.uk/Product/Robin-DiAngelo/White-Fragility--Why-Its-So-Hard-for-White-People-to-Talk-About-Racism/23286943"
          >
            White Fragility - Why It's So Hard For White People To Talk About
Racism
          </a>
           by Robin DiAngelo
        </li>
        <li>
          <a
            href="https://www.hive.co.uk/Product/Layla-Saad/Me-and-White-Supremacy--How-to-Recognise-Your-Privilege-C/24440368"
          >
            Me and White Supremacy: How to Recognise Your Privilege, Combat Racism and Change the World
          </a>
           by Layla F Saad
        </li>
        <li>
          <a
            href="https://www.hive.co.uk/Product/Ibram-X-Kendi/How-To-Be-an-Antiracist/23804402"
          >
            How To Be An Antiracist
          </a>
           by Ibram X Kendi
        </li>
        <li>
          <a
            href="https://www.hive.co.uk/Product/Ijeoma-Oluo/So-You-Want-to-Talk-About-Race/24581698"
          >
            So You Want To Talk About Race
          </a>
           by Ijeoma Oluo
        </li>
        <li>
          <a
            href="https://www.hive.co.uk/Product/Reni-Eddo-Lodge/Why-Im-No-Longer-Talking-to-White-People-About-Race--The-/21140288"
          >
            Why I'm No Longer Talking To White People About Race
          </a>
           by Reni Eddo-Lodge
        </li>
      </ul>
      <nav
        className="css-1b6p5hc"
      >
        <a
          className="flex-initial rounded-lg bg-white text-purple p-3 no-underline"
          href="/donate"
        >
          I have some spare cash.
        </a>
        <a
          className="flex-initial rounded-lg bg-white text-purple p-3 no-underline"
          href="/amplify"
        >
          I'd like to show solidarity.
        </a>
      </nav>
    </main>
    <footer
      className="flex justify-center space-x-5 css-1bxtzlt"
    >
      <div>
        <a
          href="https://thenounproject.com/cesarreynoso562/"
        >
          Fist icon by Cesar Reynoso
        </a>
      </div>
      <div>
        <a
          href="https://github.com/maccath/blm"
        >
          <svg
            className="inline mr-2"
            fill="currentColor"
            height={15}
            viewBox="0 0 24 24"
            width={15}
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              GitHub
            </title>
            <path
              d="M12 .297c-6.63 0-12 5.373-12 12 0 5.303 3.438 9.8 8.205 11.385.6.113.82-.258.82-.577 0-.285-.01-1.04-.015-2.04-3.338.724-4.042-1.61-4.042-1.61C4.422 18.07 3.633 17.7 3.633 17.7c-1.087-.744.084-.729.084-.729 1.205.084 1.838 1.236 1.838 1.236 1.07 1.835 2.809 1.305 3.495.998.108-.776.417-1.305.76-1.605-2.665-.3-5.466-1.332-5.466-5.93 0-1.31.465-2.38 1.235-3.22-.135-.303-.54-1.523.105-3.176 0 0 1.005-.322 3.3 1.23.96-.267 1.98-.399 3-.405 1.02.006 2.04.138 3 .405 2.28-1.552 3.285-1.23 3.285-1.23.645 1.653.24 2.873.12 3.176.765.84 1.23 1.91 1.23 3.22 0 4.61-2.805 5.625-5.475 5.92.42.36.81 1.096.81 2.22 0 1.606-.015 2.896-.015 3.286 0 .315.21.69.825.57C20.565 22.092 24 17.592 24 12.297c0-6.627-5.373-12-12-12"
            />
          </svg>
          Contribute on GitHub
        </a>
      </div>
    </footer>
  </div>
</div>
`;
